src/Hydra/Protocol/Figures/offchain-protocol.tex

```diff
@@ -75,35 +75,63 @@
 								$\hatmT \gets \hatmT \cup \{\tx\}$ \;
 								% issue snapshot if we are leader
 								\If{$\hats = \bar{\mc S}.s \land \hpLdr(\bar{\mc S}.s + 1) = i$}{
-									\Multi{} $(\hpRS,\hatv,\bar{\mc S}.s+1,\hatmT \red{,\tx_\alpha}, \tx_\omega)$ \;
+									\Multi{} $(\hpRS,\hatv,\bar{\mc S}.s+1,\hatmT, \red{U_\alpha}, \tx_\omega)$ \;
 								}
 							}
 						}
+						\vspace{12pt}
+
+						%%% REC DEC
+						\On{$(\mathtt{reqDec},\tx)$ from $\party_j$}{
+							\Wait{$\red{U_{\alpha} = \emptyset ~ \land ~}\tx_\omega  = \bot ~ \land ~ \hatmL \applytx \tx \ne \bot$}{
+								$\hatmL \gets \hatmL \applytx \tx \setminus \mathsf{outputs}(\tx)$ \;
+								$\tx_\omega \gets \tx$ \;
+								% issue snapshot if we are leader
+								\If{$\hats = \bar{\mc S}.s \land \hpLdr(\bar{\mc S}.s + 1) = i$}{
+									\Multi{} $(\hpRS,\hatv,\bar{\mc S}.s+1,\hatmT, \red{U_\alpha},\tx_\omega)$ \;
+								}
+							}
+						}
+						\vspace{12pt}
 
+						%%% DEPOSIT
+						\red{
+							\On{$(\mathtt{depositTx}, U)$ from chain}{
+							  % FIXME: wait on chain events is a bit weird. Wait
+							  % in general feels like avoiding book keeping and
+							  % relying a lot on assumption of a perfect queue
+							  \Wait{$\tx_\omega = \bot ~ \land ~ U_{\alpha} = \emptyset$}{
+								$U_{\alpha} = U$ \;
+								% issue snapshot if we are leader
+								\If{$\hats = \bar{\mc S}.s \land \hpLdr(\bar{\mc S}.s + 1) = i$}{
+								  \Multi{} $(\hpRS,\hatv,\bar{\mc S}.s+1,\hatmT, \red{U_\alpha}, \tx_\omega)$ \;
+								}
+							  }
+							}
+						}
 						\vspace{12pt}
 
 						%%% REQ SN
-						\On{$(\hpRS,v,s,\mT_{\mathsf{req}} \red{, \tx_\alpha} , \tx_\omega)$ from $\party_j$}{
-							\red{\Req{$\tx_\omega = \bot ~ \lor ~ \tx_\alpha = \bot$}} \;
+						\On{$(\hpRS,v,s,\underline{\tx}_{\mathsf{req}} \red{, U_\alpha} , \tx_\omega)$ from $\party_j$}{
```

Comment:
Why did we have to alter this to `_underline{\tx}`?

Reply:
I renamed $\mathcal{T}_\mathsf{req}$ to ${\underline{tx}} _\mathsf{req}$ as the latter is more consistent with the notation introduced in preliminaries.